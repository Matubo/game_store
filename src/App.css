body {
  background-image: url('./assets/img/background.jpg');
  background-position: center;
  background-attachment: fixed;
}
