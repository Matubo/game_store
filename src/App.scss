body {
  font-family: 'Silkscreen', 'Silkscreen', 'Pixel Times', 'Times New Roman', Times, serif;
  color: white;
  margin: 0;
  padding: 0;
  max-width: 100vw;
  overflow-x: hidden;
}
html {
  height: 100%;
  max-width: 100%;
  margin: 0;
  padding: 0;
}
button {
  font-size: medium;
  font-weight: bold;
  font-family: 'Silkscreen';
}
input,
select {
  font-family: 'Silkscreen', 'Pixel Times';
  font-size: medium;
  font-weight: bold;
}
.app-background {
  position: fixed;
  height: 100vh;
  min-width: 100vw;
  top: 0;
  left: 0;
  z-index: -1;
  background-image: url('./assets/img/background.jpg');
  background-position: center;
}

@media screen and (orientation: portrait) {
  .app-background {
    height: 120vh;
  }
}

p {
  font-size: small;
}
.app-loader {
  position: fixed;
  top: 0;
  left: 0;
  height: 100vh;
  width: 100vw;
  background-color: rgba(122, 121, 121, 0.562);
  display: flex;
  align-items: center;
  justify-content: center;
  img {
    max-width: 50%;
    max-height: 50%;
  }
}
