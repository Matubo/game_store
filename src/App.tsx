import { Suspense } from 'react';
import { BrowserRouter, Routes, Route, Link } from 'react-router-dom';
import './App.css';
import { routes } from './consts/routes';
import MineTemplate from './pages/MineTemplate/MineTemplate';
import AppRouter from './router/AppRouter';
import TestBlock from './TestBlock';
import './assets/fonts/font.css';

export default function App() {
  return (
    <div className="React-App">
      <BrowserRouter>
        <MineTemplate>
          <Suspense fallback={<p>Загрузка</p>}>
            <AppRouter></AppRouter>
          </Suspense>
        </MineTemplate>
        <TestBlock></TestBlock>
      </BrowserRouter>
    </div>
  );
}
