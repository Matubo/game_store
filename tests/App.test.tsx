import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import App from 'src/App';
import setupStore from 'src/redux/store';

describe('App test', () => {
  test('snapshot test', () => {
    const { asFragment } = render(
      <Provider store={setupStore()}>
        {/* надо перенести в отдельный хелпер */}
        <App></App>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
