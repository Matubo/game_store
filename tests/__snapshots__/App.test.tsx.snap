// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test snapshot test 1`] = `
<DocumentFragment>
  <div
    class="React-App"
  >
    <div
      style="display: flex; flex-direction: column; min-height: 100vh; justify-content: flex-start;"
    >
      <div
        class="navbar-header"
      >
        <a
          class="navbar-header__main-logo main-logo"
          href="/"
        >
          <img
            alt="logo"
            class="main-logo__img"
            src="[object Object]"
          />
          <p
            class="main-logo__text"
          >
            All games
          </p>
        </a>
        <a
          class="navbar-header__item header-item"
          href="/"
        >
          <p
            class="header-item__text"
          >
            home
          </p>
        </a>
        <div
          class="navbar-header__item dropdown-item"
          tabindex="0"
        >
          <p
            class="dropdown-item__link"
          >
            category
          </p>
        </div>
        <a
          class="navbar-header__item header-cart"
          href="/user"
        >
          <img
            alt="cart"
            class="header-cart__img"
            src="[object Object]"
          />
        </a>
        <a
          class="navbar-header__item header-cart"
          href="/cart"
        >
          <img
            alt="cart"
            class="header-cart__img"
            src="[object Object]"
          />
        </a>
      </div>
      <div
        class="home-page"
      >
        <p
          class="home-page__heading"
        >
          TOP DISCOUNT
        </p>
        <div
          class="search-result"
        />
        <p
          class="home-page__heading"
        >
          USER SMALL REVIEW
        </p>
        <div
          class="reviews"
          style="height: 0px;"
        />
      </div>
      <div
        class="footer"
      >
        <div
          class="footer__line"
        />
        <p
          class="footer__text"
        >
          Evergreen Terrace, 742, Springfield, USA - 555-0113
          <br />
          Â©2022 "All games"
        </p>
      </div>
    </div>
    <div
      class="app-background"
    />
  </div>
</DocumentFragment>
`;
